futures-core/src/future/mod.rs

```diff
@@ -14,6 +14,10 @@ pub use self::future_obj::{FutureObj, LocalFutureObj, UnsafeFutureObj};
 /// statically type your result or need to add some indirection.
 pub type BoxFuture<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + Send + 'a>>;
 
+#[cfg(feature = "alloc")]
+/// `BoxFuture`, but without the `Send` requirement.
+pub type BoxFutureLocal<'a, T> = Pin<alloc::boxed::Box<dyn Future<Output = T> + 'a>>;
```

Comment:
The other `Local` types use it as a prefix on the type name (`LocalFutureObj`, `LocalSpawn`, `LocalPool`) but the functions do seem to be a suffix, so maybe this should be `fn boxed_local() -> LocalBoxFuture` for consistency? (I don't know if there's other convention for this pattern outside this library?)

Reply:
@Nemo157 the type is now renamed to `LocalBoxFuture`.